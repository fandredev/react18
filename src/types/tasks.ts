export interface TasksListProps {
  task: string
  time: string
  completed: boolean
  selected: boolean
  id: string
}
