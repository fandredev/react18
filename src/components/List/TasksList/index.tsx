import classNames from 'classnames';
import { TasksListProps } from "types/tasks";
import styles from './taskslist.module.scss';

interface ListTasksProps extends TasksListProps {
  selectTask: (selectedTask: TasksListProps) => void
}


export default function TasksList({ task, time, id, selected, completed, selectTask }: ListTasksProps) {
  return (
    <li className={classNames({
      [styles.item]: true,
      [styles.listSelected]: selected,
      [styles.listCompleted]: completed
    })} onClick={() => !completed && selectTask({
      task, time, id, selected, completed
    })}>
      <h3>{task}</h3>
      <span>{time}</span>
      {completed && <span className={styles.completed} aria-label="Tarefa completada" />}
    </li>
  )
}
