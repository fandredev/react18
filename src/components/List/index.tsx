import { TasksListProps } from "types/tasks";
import styles from "./list.module.scss";
import TasksList from "./TasksList";

interface ListProps {
  tasks: TasksListProps[]
  selectTask: (selectedTask: TasksListProps) => void
}

export default function List({ tasks, selectTask }: ListProps): JSX.Element {

  return (
    <aside className={styles.listTasks}>
      {tasks.length === 0 ? <h2>Você ainda não adicionou algum estudo hoje.</h2> : <h2>Estudos do dia</h2>}
      <ul>
        {tasks.map((task) =>
          <TasksList selectTask={selectTask} {...task} key={task.id} />
        )}
      </ul>
    </aside>
  )
}
