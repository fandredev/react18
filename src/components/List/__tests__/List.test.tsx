import { render, screen } from "@testing-library/react";
import { task } from "common/mocks/List.mock";
import { TasksListProps } from "types/tasks";
import List from "..";
import TasksList from "../TasksList";

describe(`Test component: ${List.name}`, () => {
  const emptyTasks: TasksListProps[] = []

  it('Should be possible to snapshot the Test component when the page is rendered', () => {
    const { container } = render(
      <List selectTask={() => null} tasks={emptyTasks} />
    )

    expect(container).toMatchSnapshot()
  })

  it("Should be able to show the page title when i'm not a list of studies", async () => {
    render(<List selectTask={() => null} tasks={emptyTasks} />)

    const title = await screen.findByText('Você ainda não adicionou algum estudo hoje.')
    expect(title).toBeInTheDocument()
  })

  it('Should be able to show the page title when I have one or more studies added to the list', async () => {
    render(<List selectTask={() => null} tasks={task} />)

    const title = await screen.findByText('Estudos do dia')
    expect(title.textContent).toBe('Estudos do dia')
  })

  it(`It should be able to render the ${TasksList.name} component when one or more items are added to the list`, async () => {
    render(<List selectTask={() => null} tasks={task} />)

    const [itemList] = screen.queryAllByRole('listitem')
    expect(itemList).toBeInTheDocument()
  })

  it('Should be able to have an item in the list when an item is added to the list', () => {
    render(<List selectTask={() => null} tasks={task} />)

    const itemList = screen.queryAllByRole('listitem')
    expect(itemList).toHaveLength(1)
  })

  it('Should be able to have more than one item in the list when items are added to the list', () => {
    const tasks = [{
      task: 'Aprender Angular',
      id: '2',
      selected: false,
      completed: false,
      time: '01:00:00'
    }, {
      task: 'Aprender Vue',
      id: '3',
      selected: false,
      completed: false,
      time: '01:00:00'
    }]

    render(<List selectTask={() => null} tasks={tasks} />)

    const itemList = screen.queryAllByRole('listitem')
    expect(itemList).toHaveLength(2)
  })
})
