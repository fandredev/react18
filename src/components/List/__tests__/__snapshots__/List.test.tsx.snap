// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: List Should be possible to snapshot the Test component when the page is rendered 1`] = `
<div>
  <aside
    class="listTasks"
  >
    <h2>
      Você ainda não adicionou algum estudo hoje.
    </h2>
    <ul />
  </aside>
</div>
`;
